includes/wc-user-functions.php

```diff
@@ -196,54 +196,13 @@ function wc_paying_customer( $order_id ) {
  * @return bool
  */
 function wc_customer_bought_product( $customer_email, $user_id, $product_id ) {
-	global $wpdb;
-
-	$result = apply_filters( 'woocommerce_pre_customer_bought_product', null, $customer_email, $user_id, $product_id );
+	wc_deprecated_function( 'WC_Order_Product_Lookup_Helper::is_product_purchased', '3.3' );
```

Comment:
I don't know if it's wise to deprecate these, given `WC_Order_Product_Lookup_Helper::is_product_purchased` is a temporary solution.

Ping @kloon @mikejolley

Reply:
I think it would be better to continue using the functions throughout the codebase, and call the helper in here. Mark the helper as 'private' in docblocks for internal use only.